x/wasm/keeper/relay_test.go

```diff
@@ -462,17 +484,23 @@ func TestOnRecvPacket(t *testing.T) {
 				return
 			}
 			require.NoError(t, err)
-			require.Equal(t, spec.expAck, gotAck)
+			require.Equal(t, spec.expAck, gotAck.Acknowledgement())
 
 			// verify gas consumed
 			const storageCosts = sdk.Gas(2903)
 			assert.Equal(t, spec.expContractGas, ctx.GasMeter().GasConsumed()-before-storageCosts)
-			// verify msgs dispatched
-			require.Len(t, *capturedMsgs, len(spec.contractResp.Messages))
-			for i, m := range spec.contractResp.Messages {
-				assert.Equal(t, (*capturedMsgs)[i], m.Msg)
+
+			// verify msgs dispatched on success/ err response
```

Comment:
Okay, if you assert this here, you just need to return non-empty messages and events more often above to make this actually meaningful (with 0 sent, it passes in both cases)

Reply:
Yes, this is correct.
But most test cases don't dispatch any events, so these cases are not properly tested

Reply:
In `ibc_tests` the event handling fully covered. For all cases in `TestOnRecvPacket` a custom event is [emitted](https://github.com/CosmWasm/wasmd/pull/1358/files#diff-4cda9df144af40622d0fed5b47665bedfde425ceded6eece574b84eb03a424bbR105-R106) so that the behaviour can be verified. Only for success or nil Ack the `myCustomEvent` is in the expected event set.
It can be confusing as we always have the `ibc_packet_received` event for non panic cases